sway-core/src/language/ty/declaration/declaration.rs

```diff
@@ -468,6 +468,22 @@ impl TyDeclaration {
         }
     }
 
+    /// name string used in `forc doc` file path generation that mirrors `cargo doc`.
+    pub fn doc_name(&self) -> &'static str {
+        use TyDeclaration::*;
+        match self {
+            StructDeclaration(_) => "struct",
+            EnumDeclaration(_) => "enum",
+            TraitDeclaration(_) => "trait",
+            AbiDeclaration(_) => "abi",
+            StorageDeclaration(_) => "storage",
+            ImplTrait(_) => "impl_trait",
+            FunctionDeclaration(_) => "fn",
+            ConstantDeclaration(_) => "constant",
+            _ => "", // non-documentable items, this arm is unreachable
```

Comment:
If the arm is really unreachable, using `unreachable!("...")` would be nicer than just a comment

Reply:
Okay nice! Thanks for the review